engine/apps/api/views/alert_group.py

```diff
@@ -228,6 +230,30 @@ def retrieve(self, request, *args, **kwargs):
             return Response(data={"error_code": "wrong_team"}, status=status.HTTP_403_FORBIDDEN)
 
 
+class AlertGroupSearchFilter(SearchFilter):
+    def filter_queryset(self, request, queryset, view):
+        search_fields = self.get_search_fields(view, request)
+        search_terms = self.get_search_terms(request)
+        if not search_fields or not search_terms:
+            return queryset
+
+        if settings.FEATURE_ALERT_GROUP_SEARCH_CUTOFF_DAYS:
+            started_at = request.query_params.get("started_at")
+            end = DateRangeFilterMixin.parse_custom_datetime_range(started_at)[1] if started_at else timezone.now()
+            queryset = queryset.filter(
+                started_at__gte=end - timedelta(days=settings.FEATURE_ALERT_GROUP_SEARCH_CUTOFF_DAYS)
+            )
+
+        return super().filter_queryset(request, queryset, view)
+
+    def get_search_fields(self, view, request):
+        return (
+            ["=public_primary_key", "=inside_organization_number", "web_title_cache"]
```

Comment:
(maybe later) I would also consider enabling only title search (or reimplement the other field searches to be exact filters instead of introducing OR LIKE queries).